Description: Only add timestamp when BUILD_CREDIT is defined
Author: Juan Picca <jumapico@gmail.com>
Last-Update: 2015-06-23
---
Index: aegisub/src/version.cpp
===================================================================
--- aegisub.orig/src/version.cpp
+++ aegisub/src/version.cpp
@@ -56,7 +56,11 @@ const char *GetAegisubShortVersionString
 }
 
 const char *GetAegisubBuildTime() {
+#ifdef BUILD_CREDIT
 	return __DATE__ " " __TIME__;
+#else
+	return "";
+#endif
 }
 
 const char *GetAegisubBuildCredit() {
